### ISSUE TITLE
RuleRequest: no_constant_condition

### ISSUE THREAD

**Issue by lo1tuma** [realm/SwiftLint#4668]
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### New rule request

1. Why should this rule be added? Share links to existing discussion about what
   the community thinks about this.

Conditional branches where the condition constantly evaluates to `true` or `false` are useless because only one branch is executed always or never.

2. Provide several examples of what _would_ and _wouldn't_ trigger violations.

Violating Examples

```swift
if true {
    print("foo")
}

if false {
    print("foo")
}

if x && false {
    print("foo")
}

if x || true {
    print("foo")
}

switch "b" {
case "a":
    print("foo")
case "c":
    print("bar")
default:
    break
}
```

Non-Violating Examples

```swift
if x {
    print("foo")
}

switch x {
case "a":
    print("foo")
case "c":
    print("bar")
default:
    break
}
```

3. Should the rule be configurable, if so what parameters should be configurable?

This rule doesn’t need to be configurable.

4. Should the rule be opt-in or enabled by default? Why?
   See [README.md](../README.md#opt-in-rules) for guidelines on when to mark a rule as opt-in.

I think this rule would be a good candidate for the default rules since it points out a possible programming error.

**Comment by theresagrubbs1027** [comment]
> ### New Issue Checklist
> * [x]  Updated SwiftLint to the latest version
> * [x]  I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)
> 
> ### New rule request
> 1. Why should this rule be added? Share links to existing discussion about what
>    the community thinks about this.
> 
> Conditional branches where the condition constantly evaluates to `true` or `false` are useless because only one branch is executed always or never.
> 
> 2. Provide several examples of what _would_ and _wouldn't_ trigger violations.
> 
> Violating Examples
> 
> ```swift
> if true {
>     print("foo")
> }
> 
> if false {
>     print("foo")
> }
> 
> if x && false {
>     print("foo")
> }
> 
> if x || true {
>     print("foo")
> }
> 
> switch "b" {
> case "a":
>     print("foo")
> case "c":
>     print("bar")
> default:
>     break
> }
> ```
> 
> Non-Violating Examples
> 
> ```swift
> if x {
>     print("foo")
> }
> 
> switch x {
> case "a":
>     print("foo")
> case "c":
>     print("bar")
> default:
>     break
> }
> ```
> 
>  3. Should the rule be configurable, if so what parameters should be configurable?
> 
> This rule doesn’t need to be configurable.
> 
> 4. Should the rule be opt-in or enabled by default? Why?
>    See [README.md](../README.md#opt-in-rules) for guidelines on when to mark a rule as opt-in.
> 
> I think this rule would be a good candidate for the default rules since it points out a possible programming error.

Theresa troubles
